app/models/publisher.rb

```diff
@@ -239,6 +240,36 @@ def dont_destroy_verified_publishers
     throw :abort if verified?
   end
 
+  # Returns publishers who have abandoned verification process and do not have another publisher account
+  def self.get_winback_publishers
+    verified_publishers = Publisher.where(verified: true)
+
+    # Only include publishers who have filled contact information
+    unverified_publishers = Publisher.where(verified: false).where.not(brave_publisher_id: [nil]).where("created_at < ?", WINBACK_THRESHOLD.ago)
+
+    winback_publishers = []
+    unverified_publishers.each do |publisher|
```

Comment:
Similarly `.each` here forces all records matching the query to be loaded into memory at once. Using `find_each` is probably preferable, as it will only load 1000 records into memory at once: http://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each